### PR TITLE
Allow different expires timestamp

### PR DESCRIPTION
**Pull request by markusramsak**

### CODE REVIEW

**Comment by lorenzo** [comment]
What is this for?

**Comment by mdlouhy** [comment]
the gc() method has an "expires" param -> it should be used.

**Comment by lorenzo** [comment]
Actually, I think `gc` is a method that should be removed. It only exists in the FileEngine class

**Comment by mdlouhy** [comment]
that's an option too

**Comment by ADmad** [comment]
There's an empty stub method `CacheEngine::gc()` overridden only in `FileEngine`.

**Comment by bcrowe** [comment]
Rather than removing `gc()`, what about offering garbage collection in the other engines?

**Comment by lorenzo** [comment]
@bcrowe there is rarely a meaningful sense for garbage collection in other engines, specially for those that are memory based with automatic key eviction (Memcached, Redis, Apc).

**Comment by mdlouhy** [comment]
it is not that painful to write ->clear(false) instead of gc() therefore I suggest to remove gc method

**Comment by ADmad** [comment]
:+1: to killing `gc()`

**Comment by lorenzo** [comment]
@mdlouhy do you want to change this PR to remove that method?

File: src/Cache/Engine/FileEngine.php [hunk 1]
```diff
@@ -103,17 +103,6 @@ public function init(array $config = [])
     }
 
     /**
-     * Garbage collection. Permanently remove all expired and deleted data
-     *
-     * @param int|null $expires [optional] An expires timestamp, invalidating all data before.
-     * @return bool True if garbage collection was successful, false on failure
-     */
-    public function gc($expires = null)
```
Comment (markstory) [inline src/Cache/Engine/FileEngine.php]: If we're removing this method, we should also consider removing `Cache::gc()`, and the call to gc() when engines are created by CacheRegistry.

Comment (markstory) [inline src/Cache/Engine/FileEngine.php]: The main purpose of gc() is to garbage collect filesystem caches, which do not have a way to vacuum expired data. We need to retain that behavior or file caching will eventually result in disk exhaustion.

Comment (jadb) [inline src/Cache/Engine/FileEngine.php]: Reminds me of this by @AD7six:

http://ad7six.com/blog/2015/01/07/file-caching-preventing-running-out-of-space/

**Comment by mdlouhy** [comment]
I think gc() should be replaced by user cronjobs which call Cache::clear(false)
And in the documentation we could add an info how to do it.

**Comment by markstory** [comment]
@mdlouhy I think expecting people to add cron jobs is not really that reasonable. File caching is often used in deployment scenarios where you don't have access to better caching engines. Forcing people who are likely to be in less than ideal deployment environments to also setup cron jobs seems unwise.

**Comment by ADmad** [comment]
Given @markstory's observations it would be wise not to remove gc()

**Comment by jadb** [comment]
:+1:  for keeping `gc()`.

**Comment by ADmad** [comment]
Closing as the new consensus seems to be keep this method. If someone figures out how to use the `$expires` param a new PR can be made.

**Pull request closed without merging**